File: src/chipschallenge/Move.java
package chipschallenge;

import java.awt.Point;

/**
 *
 * @author patrik
 */
public class Move {

    public enum Moves {UP, DOWN, LEFT, RIGHT};

    public static void updatePoint(Point p, Moves m) {
        switch (m) {
            case UP:
                p.y--;
                break;
            case DOWN:
                p.y++;
                break;
            case LEFT:
                p.x--;
                break;
            case RIGHT:
                p.x++;
                break;
        }
    }

    public static Moves getRandom() {
        int random = Utils.r.nextInt(4);
        Moves ret = null;
        switch(random) {
            case 0:
                ret = Moves.UP;
                break;
            case 1:
                ret = Moves.DOWN;
                break;
            case 2:
                ret = Moves.LEFT;
                break;
            case 3:
                ret = Moves.RIGHT;
                break;
        }
        return ret;
    }

    public static Moves reverse(Moves m) {
        Moves ret = null;
        switch(m) {
            case UP:
                ret = Moves.DOWN;
                break;
            case DOWN:
                ret = Moves.UP;
                break;
            case LEFT:
                ret = Moves.RIGHT;
                break;
            case RIGHT:
                ret = Moves.LEFT;
                break;
        }
        return ret;
    }
    
}

File: ChipsChallenge.Shared/Move.cs
// ReSharper disable InconsistentNaming
namespace ChipsChallenge.Shared
{
    public class Move
    {
        public enum Moves
        {
            UP,
            DOWN,
            LEFT,
            RIGHT
        }

        public static void UpdatePoint(ref Point p, Moves m)
        {
            switch (m)
            {
                case Moves.UP:
                    p.Y--;
                    break;
                case Moves.DOWN:
                    p.Y++;
                    break;
                case Moves.LEFT:
                    p.X--;
                    break;
                case Moves.RIGHT:
                    p.X++;
                    break;
            }
        }

        public static Moves Random
        {
            get
            {
                int random = Utils.Random.Next(4);

                switch (random)
                {
                    case 0:
                        return Moves.UP;
                    case 1:
                        return Moves.DOWN;
                    case 2:
                        return Moves.LEFT;
                    case 3:
                        return Moves.RIGHT;
                    default:
                        return Moves.UP;
                }
            }
        }

        public static Moves Reverse(Moves m)
        {
            switch (m)
            {
                case Moves.UP:
                    return Moves.DOWN;
                case Moves.DOWN:
                    return Moves.UP;
                case Moves.LEFT:
                    return Moves.RIGHT;
                case Moves.RIGHT:
                    return Moves.LEFT;
                default:
                    return Moves.UP;
            }
        }
    }
}